design-documents/frontend/technical-vision.md

```diff
@@ -0,0 +1,140 @@
+# Front-End Technical Vision
+
+The following document is a guide for all future Magento front-end applications, for which we should be mindful of the following values:
+
+- Portability
+- Extensibility
+- Maintainability
+- Consistency
+- Performance
+- Accessibility
+- Internationalization
+- Productivity (Efficiency)
+- Quality (Reliability)
+
+## [React](https://reactjs.org/)
+
+_Portability, Performance, Productivity, Quality, Internationalization._
+
+React's [continued growth and popularity](https://www.npmtrends.com/@angular/core-vs-react-vs-vue) coupled with its [solid documentation](https://reactjs.org/docs/getting-started.html) make it an excellent choice for building user interfaces. If written appropriately, it enables **portability** by [crossing over into native applications](http://jkaufman.io/react-web-native-codesharing/) with [React Native](https://facebook.github.io/react-native/).
+
+**Performance** is easy to achieve with React, because it only re-renders DOM elements where and when necessary. That's not to say that React prevents you from creating **performance** problems, but the docs do well at explaining [what to avoid and how to profile](https://reactjs.org/docs/optimizing-performance.html).
+
+Creating a new React project has never been easier with [CRA][], which comes out-of-the-box with [hot reloading](https://webpack.js.org/concepts/hot-module-replacement/). This increases **productivity** by giving developers [short, positive feedback loops](https://www.youtube.com/watch?v=avKBrgIQXzk). Hot reloading is [also achievable in React Native](https://facebook.github.io/react-native/blog/2016/03/24/introducing-hot-reloading.html). Another opportunity for short, positive feedback loops is testing with [Jest][] and [Enzyme](https://airbnb.io/enzyme/). Tests are fastest when writing unit tests with the [Shallow Rendering API](https://airbnb.io/enzyme/docs/api/shallow.html) and reserving the [Full Rendering API](https://airbnb.io/enzyme/docs/api/mount.html) for integration tests. The better the development experience in these cycles, the better **quality** we achieve as a result. [Jest's snapshot testing](https://jestjs.io/docs/en/snapshot-testing) is another way to ensure **quality** in that we don't inadvertently introduce breaking changes.

```

Comment:
`[CRA][]` here is missing the link value

Reply:
Look at line 138.